drift/views/v1.py

```diff
@@ -285,6 +287,30 @@ def get_historical_sys_profiles(
                             str(HTTPStatus.FORBIDDEN) + " " + message, request=request
                         )
                         raise HTTPError(HTTPStatus.FORBIDDEN, message=message)
+                    except HTTPError as error:
+                        http_errors.append(error)
+                    except (ItemNotReturned, ServiceError) as error:
+                        other_errors.append(error)
+
+            if http_errors or other_errors:
+                if any(type(other_error) == ServiceError for other_error in other_errors) or any(
+                    http_error.code >= 500 for http_error in http_errors
+                ):
+                    http_code = HTTPStatus.INTERNAL_SERVER_ERROR
+                elif any(http_error.code == HTTPStatus.BAD_REQUEST for http_error in http_errors):
+                    http_code = HTTPStatus.BAD_REQUEST
+                elif any(
+                    type(other_error) == ItemNotReturned for other_error in other_errors
+                ) or any(http_error.code == HTTPStatus.NOT_FOUND for http_error in http_errors):
+                    http_code = HTTPStatus.NOT_FOUND
+                else:
+                    http_code = HTTPStatus.INTERNAL_SERVER_ERROR
+
+                message = ", ".join(
+                    [http_error.reason for http_error in http_errors]
```

Comment:
I'd suggest to add `str()` to be sure that the join would join strings.

Reply:
Good suggestion!